torchvision/csrc/cpu/DeformConv_cpu.cpp

```diff
@@ -232,22 +232,23 @@ at::Tensor DeformConv2d_forward_cpu(
     const at::Tensor& input_param,
     const at::Tensor& weight_param,
     const at::Tensor& offset_param,
-    const at::Tensor& bias,
-    std::pair<int, int> stride,
-    std::pair<int, int> pad,
-    std::pair<int, int> dilation,
-    int n_weight_grps,
-    int n_offset_grps) {
-  at::Tensor input = input_param;
-  at::Tensor offset = offset_param;
-  at::Tensor weight = weight_param;
+    const at::Tensor& bias_param,
+    int64_t stride_h,
+    int64_t stride_w,
+    int64_t pad_h,
+    int64_t pad_w,
+    int64_t dil_h,
+    int64_t dil_w,
+    int64_t n_weight_grps,
+    int64_t n_offset_grps) {
+  at::Tensor&& input = input_param.contiguous();
```

Comment:
This is not something that I generally see being done within the PyTorch codebase.

For example I'd generally see [this in PyTorch](https://github.com/pytorch/pytorch/blob/daf2a6a29da5109157bb44fc5e4dbc4fb5c56c15/aten/src/ATen/native/ConvolutionMM2d.cpp#L128)

Reply:
Eh, it's pretty common in PyTorch codebase, if the kernel is written in such a way that it only works for contiguous inputs, you'll call contiguous first thing in the kernel. In fact, even in the linked code code you quoted is doing a contiguous call (there is just some extra gobbeldygook to reshape if there's batching involved).

Reply:
The reason why it's wrong to call contiguous before the dispatcher call, is because you may enter the dispatcher directly via TorchScripted code. Of course you can call a TorchScripted function with non-contiguous inputs, so you had better have handling for the non-contiguous case somewhere!

Reply:
Oh, I think there was a misunderstanding here. I was referring to the `&&`.

Reply:
I removed the rvalue references.